pkg/gardenlet/controller/seed/seed/components.go

```diff
@@ -319,10 +319,12 @@ func (r *Reconciler) newIstio(ctx context.Context, seed *seedpkg.Seed, isGardenC
 
 	servicePorts := []corev1.ServicePort{
 		{Name: "tcp", Port: 443, TargetPort: intstr.FromInt32(9443)},
-		{Name: "http-proxy", Port: vpnseedserver.HTTPProxyGatewayPort, TargetPort: intstr.FromInt(vpnseedserver.HTTPProxyGatewayPort)},
 		// TODO(hown3d): Drop with RemoveHTTPProxyLegacyPort feature gate
 		{Name: "tls-tunnel", Port: vpnseedserver.GatewayPort, TargetPort: intstr.FromInt32(vpnseedserver.GatewayPort)},
 	}
+	if features.DefaultFeatureGate.Enabled(features.UseUnifiedHTTPProxyPort) {
+		servicePorts = append(servicePorts, corev1.ServicePort{Name: "http-proxy", Port: vpnseedserver.HTTPProxyGatewayPort, TargetPort: intstr.FromInt(vpnseedserver.HTTPProxyGatewayPort)})
+	}
```

Comment:
Is it possible to cover this with unit test?

Reply:
As there are no unit tests at all yet, I would rather see it as follow up and not add it as part of a small bug fix.
WDYT?

Reply:
> As there are no unit tests at all yet, I would rather see it as follow up and not add it as part of a small bug fix. WDYT?

OK. It won't be fair against you to ask you to add unit tests if the previous feature PR did not add unit tests.
Could you open a follow-up issue like https://github.com/gardener/gardener/issues/13161 so that we don't forget the follow-up item?